src/components/pricing/select-plan-new/index.tsx

```diff
@@ -0,0 +1,242 @@
+      />
+    </svg>
+  )
+
+  return (
+    <ul>
+      {planFeatures.map((feature: string) => {
+        return (
+          <li className="py-2 font-medium flex" key={slugify(feature)}>
+            <CheckIcon />
+            <span className="ml-2 leading-tight">{feature}</span>
+          </li>
+        )
+      })}
+    </ul>
+  )
+}
+
+const GetAccessButton: React.FunctionComponent<{
+  label: string
+  handleClick: () => void
+}> = ({label, handleClick}) => {
+  return (
+    <button
+      className="mt-8 px-5 py-4 text-center bg-blue-600 text-white font-semibold rounded-md w-full hover:bg-blue-700 transition-all duration-300 ease-in-out transform hover:scale-105"
+      onClick={handleClick}
+      type="button"
+    >
+      {label}
+    </button>
+  )
+}
+
+type SelectPlanProps = {
+  prices: any
+  pricesLoading: boolean
+  defaultInterval?: string
+  defaultQuantity?: number
+  handleClickGetAccess: () => void
+  quantityAvailable: boolean
+}
+
+const SelectPlanNew: React.FunctionComponent<SelectPlanProps> = ({
+  quantityAvailable = true,
+  handleClickGetAccess,
+  defaultInterval = 'annual',
+  defaultQuantity = 1,
+  pricesLoading,
+  prices,
+}) => {
+  const individualPlans = filter(
+    prices,
+    (plan: any) => plan.type === 'individual',
+  )
+  const teamPlan = find(prices, {type: 'team'})
+
+  const annualPlan = get(prices, 'annualPrice')
+  const monthlyPlan = get(prices, 'monthlyPrice')
+  const quarterlyPlan = get(prices, 'quarterlyPrice')
+
+  const pricesForInterval = (interval: any) => {
+    switch (interval) {
+      case 'year':
+        return annualPlan
+      case 'month':
+        return monthlyPlan
+      case 'three months':
+        return quarterlyPlan
+      default:
+        return annualPlan
+    }
+  }
+
+  const [currentInterval] = React.useState<string>(defaultInterval)
+  const [currentQuantity, setCurrentQuantity] = React.useState<number>(
+    defaultQuantity,
+  )
+
+  const [currentPlan, setCurrentPlan] = React.useState<any>(
+    pricesForInterval(currentInterval),
+  )
+
+  const forTeams: boolean = currentQuantity >= 5
+  const buttonLabel: string = forTeams ? 'Level Up My Team' : 'Get Access'
+
+  React.useEffect(() => {
+    forTeams ? setCurrentPlan(teamPlan) : setCurrentPlan(annualPlan)
+  }, [forTeams])
+
+  return (
+    <>
+      <div className="dark:text-white text-gray-900 dark:bg-gray-900 bg-white sm:p-10 p-5 flex flex-col items-center max-w-sm relative z-10 rounded-sm">
+        <PlanTitle>{currentPlan.name}</PlanTitle>
+        <PlanPrice pricesLoading={pricesLoading} plan={currentPlan} />
+        <div className={quantityAvailable ? '' : 'pb-4'}>
+          <PlanIntervalsSwitch
+            disabled={forTeams}
+            currentPlan={currentPlan}
+            setCurrentPlan={setCurrentPlan}
+            planTypes={individualPlans}
+          />
+        </div>
+        {quantityAvailable && (
+          <div className="py-4">
```

Comment:
suggestion: spacing and basic structural layout is likely better suited for the "parent" where the component itself only defines essential styles. It's more verbose but also means the components are more flexible and you don't have to dig around sub-components for simple layout manipulation